src/gui/window.rs

```diff
@@ -47,82 +52,254 @@ impl Fps {
-            "D:/Midis/The Quarantine Project.mid",
-            SimpleTemporaryPlayer::new(),
-        ));
-
-        midi_file.timer_mut().play();
-
         GuiWasabiWindow {
             render_scene: GuiRenderScene::new(renderer),
             keyboard_layout: keyboard_layout::KeyboardLayout::new(&Default::default()),
             keyboard: GuiKeyboard::new(),
-            midi_file,
+            midi_file: None,
             fps: Fps::new(),
         }
     }
 
     /// Defines the layout of our UI
-    pub fn layout(&mut self, state: &mut GuiState) {
+    pub fn layout(
+        &mut self,
+        state: &mut GuiState,
+        perm_settings: &mut WasabiPermanentSettings,
+        temp_settings: &mut WasabiTemporarySettings,
+    ) {
         let ctx = state.gui.context();
-
+        let window_size = vec![ctx.available_rect().width(), ctx.available_rect().height()];
         self.fps.update();
-
         ctx.set_visuals(Visuals::dark());
+        let note_speed = perm_settings.note_speed;
 
-        // Render the top panel
-        egui::TopBottomPanel::top("Top panel")
-            .height_range(100.0..=100.0)
-            .show(&ctx, |ui| {
-                if let Some(length) = self.midi_file.midi_length() {
-                    let time = self.midi_file.timer().get_time().as_secs_f64();
-                    let progress = (time / length) as f32;
-                    ui.add(egui::ProgressBar::new(progress));
-                }
+        // Render the settings window if the value from
+        // the temporary settings allows it
+        if temp_settings.settings_visible {
+            egui::Window::new("Settings")
+                .resizable(true)
+                .collapsible(true)
+                .title_bar(true)
+                .scroll2([false, true])
+                .enabled(true)
+                .open(&mut temp_settings.settings_visible)
+                .show(&ctx, |ui| {
+                    egui::Grid::new("settings_grid")
+                        .num_columns(2)
+                        .spacing([40.0, 4.0])
+                        .striped(true)
+                        .show(ui, |ui| {
+                            ui.label("SFZ Path: ");
+                            ui.horizontal(|ui| {
+                                ui.add(egui::TextEdit::singleline(&mut perm_settings.sfz_path));
+                                if ui.button("Browse...").clicked() {
+                                    let sfz_path = FileDialog::new()
+                                        .add_filter("sfz", &["sfz"])
+                                        .set_directory("/")
+                                        .pick_file();
 
-                ui.add(Label::new(format!("FPS: {}", self.fps.get_fps().round())));
+                                    if let Some(sfz_path) = sfz_path {
+                                        perm_settings.sfz_path =
+                                            sfz_path.into_os_string().into_string().unwrap();
```

Comment:
We never know if there's some weird edge case where this might crash. For now we can ignore it, up to you if you're bothered patching it, but generally as I say it's good to avoid unwraps everywhere.

Mostly we can ignore because we'll probably use different file dialogue code later on, so we don't have to be too strict about this yet.